src/components/pages/movie-page/movie-page.js

```diff
@@ -36,32 +57,32 @@ const MoviePage = () => {
             </div>
             <div className="movie-cast">
               <div className="movie-cast_header">
-                <div className="movie-cast_title">Top Billed Cast</div>
-                <Button variant="outlined" size="small" color="inherit">Show all</Button>
+                <div className="movie-cast_title">{lang === 'en' ? 'Top Billed Cast' : 'Популярные актёры'}</div>
+                <Button variant="outlined" size="small" color="inherit" onClick={togglelCastItems}>{isShortListCast ? 'Show all' : 'Hide all'}</Button>
               </div>
               <ErrorBoundary>
-                <ActorList />
+                {castStatus === 'loading' && <SpinnerCircularFixed style={{display: 'block', margin: '40px auto'}}/>}
+                <ActorList data={isShortListCast ? shortListCast : movieCast}/>
               </ErrorBoundary>
             </div>
             <div className="movie-images">
-              <div className="movie-images_title">Images</div>
+              <div className="movie-images_title">{lang === 'en' ? 'Images' : 'Кадры фильма'}</div>
               <div className="movie-images_list">
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
-                <img src={image} alt="cadr from movie" />
+                {imagesStatus === 'loading' && <SpinnerCircularFixed style={{display: 'block', margin: '40px auto'}}/>}
+                {
+                  movieImages.slice(0,8).map((el, i) => 
+                    <img src={el.file_path ? `${img_300}${el.file_path}` : notfound_300} alt={'cadr from movie'} key={i}/>
+                  )
+                }
               </div>
             </div>
           </div>
         </div>
         <div>
-          <div className="movie-reccomend">Recommendations</div>
+          <div className="movie-reccomend">{lang === 'en' ? 'Recommendations' : 'Рекомендации'}</div>
           <ErrorBoundary>
-            <MovieList />
+            {recommendStatus === 'loading' && <SpinnerCircularFixed style={{display: 'block', margin: '40px auto'}}/>}
+            <MovieList data={movieRecommend.slice(0, 5)}/>
```

Comment:
each time you will provide a new array as a prop and react will make extra rerendering
for better performance calculate it inside selector or useMemo

Reply:
Fixed!